plan/logical_plan_builder.go

```diff
@@ -1827,8 +1827,8 @@ func (b *planBuilder) buildUpdate(update *ast.UpdateStmt) LogicalPlan {
 		OrderedList: orderedList,
 		IgnoreErr:   update.IgnoreErr,
 	}.init(b.ctx)
-	setParentAndChildren(updt, p)
 	updt.SetSchema(p.Schema())
+	updt.SelectPlan, b.err = doOptimize(b.optFlag, p, b.ctx)
```

Comment:
check b.err

Reply:
If b.err is not nil, we can also return. It's no need to check this.